pkg/test/client.go

```diff
@@ -0,0 +1,100 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+// NewFakeClient creates a fake K8s client with ability to override specific Get/List/Create/Update/StatusUpdate/Delete functions
+func NewFakeClient(t *testing.T, initObjs ...runtime.Object) *FakeClient {
```

Comment:
can't this project `toolchain-operator` have dependency on `toolchain-common` .. why we are not re-using fake_client from common https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/test/client.go#L14

Reply:
Because we still using kube 1.13 in common. And there are breaking API changes between 1.13 and 1.14.

When we update common we will switch to it. Right now we can’t do it because of Kubefed which still depends on 1.13

Reply:
@alexeykazakov thanks for clarification :-)